int64 degree0
int64 degree1
int64 degree2
int64 degree3
int64 degree4
int64 degree5
